test/test_cpp_api_parity.py

```diff
@@ -1,687 +1,44 @@
+PARITY_TABLE_PATH = os.path.join(os.path.dirname(__file__), 'cpp_api_parity', 'parity-tracker.md')
 
-        def add_ctor_args_test_for_module(module_name, has_impl_parity):
-            ctor_args_test_name = 'test_torch_nn_{}_ctor_args'.format(module_name)
+parity_table = parse_parity_tracker_table(PARITY_TABLE_PATH)
 
-            def ctor_args_test(self):
-                self._test_torch_nn_module_ctor_args(
-                    module_name=self._testMethodName.replace('test_torch_nn_', '').replace('_ctor_args', ''))
-
-            if not has_impl_parity:
-                ctor_args_test = unittest.expectedFailure(ctor_args_test)
-
-            # We only run one constructor args test per module
-            if not has_test(ctor_args_test_name):
-                add_test(ctor_args_test_name, ctor_args_test)
-
-        def add_variant_test_for_module(module_name, test_params_dict, has_impl_parity):
-            module_metadata = torch_nn_modules.module_metadata_map[module_name]
-            for device in devices:
-                test_params = _process_test_params(
-                    test_params_dict=test_params_dict,
-                    module_metadata=module_metadata,
-                    device=device,
-                    is_criterion=is_criterion)
-                test_name = 'test_torch_nn_{}'.format(test_params.module_variant_name)
-                torch_nn_test_params_map[test_name] = test_params
-
-                def test_fn(self):
-                    self._test_torch_nn_module_variant(test_params=torch_nn_test_params_map[self._testMethodName])
-
-                if device == 'cuda':
-                    test_fn = unittest.skipIf(not TEST_CUDA, "CUDA unavailable")(test_fn)
-
-                if not has_impl_parity:
-                    test_fn = unittest.expectedFailure(test_fn)
-
-                add_test(test_name, test_fn)
-
-        add_ctor_args_test_for_module(module_name, has_impl_parity)
-        add_variant_test_for_module(module_name, test_params_dict, has_impl_parity)
-
-add_torch_nn_module_tests(
-    sample_module.module_tests + common_nn.module_tests + common_nn.new_module_tests,
-    is_criterion=False)
-
-add_torch_nn_module_tests(
-    common_nn.criterion_tests + common_nn.new_criterion_tests,
-    is_criterion=True)
-
-# Assert that there exists auto-generated tests for SampleModule.
+class TestCppApiParity(common.TestCase):
+    pass
+
+for test_params_dicts, test_instance_class in [
+    (sample_module.module_tests, common_nn.ModuleTest),
+    (sample_functional.functional_tests, common_nn.NewModuleTest),
+    (common_nn.module_tests, common_nn.ModuleTest),
+    (common_nn.new_module_tests, common_nn.NewModuleTest),
+    (common_nn.criterion_tests, common_nn.CriterionTest),
+    (common_nn.new_criterion_tests, common_nn.NewCriterionTest),
+]:
+    module_impl_check.add_tests(TestCppApiParity, test_params_dicts, test_instance_class, parity_table, devices)
```

Comment:
Inside each of these, there is a check 
```
if 'FunctionalModule' in str(test_params_dict.get('constructor', '')):
            continue
```
that makes it so that for each item, the item only gets added to one of `module_impl_check` and `functional_impl_check`. Could we put that check here to avoid duplicating the check?

Reply:
Thanks for the catch! Fixed :D